Seem unrelated failed 
```sh
self = <tests.cli.commands.test_task_command.TestCliTasks testMethod=test_local_run>
    def test_local_run(self):
        args = self.parser.parse_args([
            'tasks',
            'run',
            'example_python_operator',
            'print_the_context',
            '2018-04-27T08:39:51.298439+00:00',
            '--interactive',
            '--subdir',
            '/root/dags/example_python_operator.py'
        ])
    
        dag = get_dag(args.subdir, args.dag_id)
        reset(dag.dag_id)
    
>       task_command.task_run(args)
tests/cli/commands/test_task_command.py:237: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
airflow/cli/commands/task_command.py:183: in task_run
    _run_task_by_selected_method(args, dag, ti)
airflow/cli/commands/task_command.py:63: in _run_task_by_selected_method
    _run_task_by_executor(args, dag, ti)
airflow/cli/commands/task_command.py:99: in _run_task_by_executor
    executor.end()
airflow/executors/local_executor.py:326: in end
    self.impl.end()
airflow/executors/local_executor.py:285: in end
    self.queue.join()
<string>:2: in join
    ???
/usr/local/lib/python3.6/multiprocessing/managers.py:757: in _callmethod
    kind, result = conn.recv()
/usr/local/lib/python3.6/multiprocessing/connection.py:250: in recv
    buf = self._recv_bytes()
/usr/local/lib/python3.6/multiprocessing/connection.py:407: in _recv_bytes
    buf = self._recv(4)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
self = <multiprocessing.connection.Connection object at 0x7f08c54e6c18>
size = 4, read = <built-in function read>
    def _recv(self, size, read=_read):
        buf = io.BytesIO()
        handle = self._handle
        remaining = size
        while remaining > 0:
            chunk = read(handle, remaining)
            n = len(chunk)
            if n == 0:
                if remaining == size:
>                   raise EOFError
E                   EOFError
/usr/local/lib/python3.6/multiprocessing/connection.py:383: EOFError
```

--- a/UPDATING.md
+++ b/UPDATING.md
@@ -61,6 +61,11 @@ https://developers.google.com/style/inclusive-documentation
 
 -->
 
+### Remove SQL support in base_hook
+
+Remove ``get_records`` and ``get_pandas_df`` and ``run`` from base_hook, which only apply for sql like hook,
+If want to use them, or your custom hook inherit them, please use ``dbapi_hook``
+
 ### Changes to SalesforceHook
 
 Replace parameter ``sandbox`` with ``domain``. According to change in simple-salesforce package

--- a/airflow/hooks/base_hook.py
+++ b/airflow/hooks/base_hook.py
@@ -75,18 +75,3 @@ class BaseHook(LoggingMixin):
     def get_conn(self):
         """Returns connection for the hook."""
         raise NotImplementedError()
-
-    def get_records(self, sql):
-        """Returns records for the sql query (for hooks that support SQL)."""
-        # TODO: move it out from the base hook. It belongs to some common SQL hook most likely
-        raise NotImplementedError()
-
-    def get_pandas_df(self, sql):
-        """Returns pandas dataframe for the sql query (for hooks that support SQL)."""
-        # TODO: move it out from the base hook. It belongs to some common SQL hook most likely
-        raise NotImplementedError()
-
-    def run(self, sql):
-        """Runs SQL query (for hooks that support SQL)."""
-        # TODO: move it out from the base hook. It belongs to some common SQL hook most likely
-        raise NotImplementedError()

--- a/airflow/providers/apache/hive/hooks/hive.py
+++ b/airflow/providers/apache/hive/hooks/hive.py
@@ -30,6 +30,7 @@ import unicodecsv as csv
 from airflow.configuration import conf
 from airflow.exceptions import AirflowException
 from airflow.hooks.base_hook import BaseHook
+from airflow.hooks.dbapi_hook import DbApiHook
 from airflow.security import utils
 from airflow.utils.helpers import as_flattened_list
 from airflow.utils.operator_helpers import AIRFLOW_VAR_NAME_FORMAT_MAPPING
@@ -775,7 +776,7 @@ class HiveMetastoreHook(BaseHook):
             return False
 
 
-class HiveServer2Hook(BaseHook):
+class HiveServer2Hook(DbApiHook):
     """
     Wrapper around the pyhive library
 
@@ -786,15 +787,15 @@ class HiveServer2Hook(BaseHook):
     are using impala you may need to set it to false in the
     ``extra`` of your connection in the UI
     """
-    def __init__(self, hiveserver2_conn_id='hiveserver2_default'):
-        super().__init__()
-        self.hiveserver2_conn_id = hiveserver2_conn_id
+    conn_name_attr = 'hiveserver2_conn_id'
+    default_conn_name = 'hiveserver2_default'
+    supports_autocommit = False
 
     def get_conn(self, schema=None):
         """
         Returns a Hive connection object.
         """
-        db = self.get_connection(self.hiveserver2_conn_id)
+        db = self.get_connection(self.hiveserver2_conn_id)  # pylint: disable=no-member
         auth_mechanism = db.extra_dejson.get('authMechanism', 'NONE')
         if auth_mechanism == 'NONE' and db.login is None:
             # we need to give a username
@@ -809,7 +810,7 @@ class HiveServer2Hook(BaseHook):
             self.log.warning(
                 "Detected deprecated 'GSSAPI' for authMechanism "
                 "for %s. Please use 'KERBEROS' instead",
-                self.hiveserver2_conn_id
+                self.hiveserver2_conn_id    # pylint: disable=no-member
             )
             auth_mechanism = 'KERBEROS'
 
@@ -833,7 +834,7 @@ class HiveServer2Hook(BaseHook):
             cur.arraysize = fetch_size or 1000
 
             # not all query services (e.g. impala AIRFLOW-4434) support the set command
-            db = self.get_connection(self.hiveserver2_conn_id)
+            db = self.get_connection(self.hiveserver2_conn_id)    # pylint: disable=no-member
             if db.extra_dejson.get('run_set_variable_statements', True):
                 env_context = get_context_from_env_var()
                 if hive_conf:

--- a/airflow/providers/grpc/hooks/grpc.py
+++ b/airflow/providers/grpc/hooks/grpc.py
@@ -97,6 +97,9 @@ class GrpcHook(BaseHook):
         return channel
 
     def run(self, stub_class, call_func, streaming=False, data=None):
+        """
+        Call gRPC function and yield response to caller
+        """
         if data is None:
             data = {}
         with self.get_conn() as channel:

